### ISSUE TITLE
Zendesk crashes app on initialize

### ISSUE THREAD

**Issue by bobvoorneveld** [zendesk/sdk_core_utilities_ios#1]
I've recently updated my SPM for ZendeskMessaging from 2.12.0 to 2.13.0.

Several other packages updated with that update and made my app crash on zendesk initialisation.

I've tried changing several packages and found the error in ZendeskSDKCoreUtilities 1.4.0, pinning that version to 1.3.0 (and thus ZendeskMessaging back to 2.12.0) fixed the issue.

This is what I call:
`Zendesk(withChannelKey: "<our key>", messagingFactory: DefaultMessagingFactory()) { result in ... }`

The error that I got:

![](https://support.zendesk.com/hc/user_images/snBCaQR90HGZIVc5Sc8_xw.png)

Stack trace:

![](https://support.zendesk.com/hc/user_images/xb1gDv99bHigl0chaH28qg.png)

**Comment by IAmConorNolan** [comment]
Hi @bobvoorneveld,

Could you tell us the version of Zendesk packages that were used with Messaging 2.13.0? Sending a screenshot of the "Package Dependencies" in Xcode would be ideal.

**Comment by bobvoorneveld** [comment]
@IAmConorNolan, these are the package installed.

![Scherm­afbeelding 2023-07-24 om 15 53 49](https://github.com/zendesk/sdk_core_utilities_ios/assets/6303073/5f8a70fa-4194-4e19-85bf-5d3b238d5bff)
![Scherm­afbeelding 2023-07-24 om 15 55 23](https://github.com/zendesk/sdk_core_utilities_ios/assets/6303073/af94fcf9-9c3f-4e67-99dd-ce30af76f5b6)

**Comment by IAmConorNolan** [comment]
Thank you for that information :)

I've tested with those versions and they are running okay for me. Is it possible the versions that caused the crash are different from the ones you listed above?

Taking a look here, it's possible we accidentally shipped a breaking change in the latest version of Core Utiltiies.

Could you confirm if the following is accurate?
1. Messaging 2.13.0 works fine with Core Utilities 1.4.0
2. Messaging 2.12.0 produces the same crash you are experiencing with Core Utilities 1.4.0

**Comment by bobvoorneveld** [comment]
I've checked my commits, this we're the installed packages:

```
{
  "pins" : [
    {
      "identity" : "sdk_conversation_kit_ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/zendesk/sdk_conversation_kit_ios",
      "state" : {
        "revision" : "bea23b0a9aaf4575e9c9656bccbea3738865fc22",
        "version" : "1.9.0"
      }
    },
    {
      "identity" : "sdk_core_utilities_ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/zendesk/sdk_core_utilities_ios",
      "state" : {
        "revision" : "3336f48956a53b37b8b7af7f514433c5ad872c5d",
        "version" : "1.4.0"
      }
    },
    {
      "identity" : "sdk_faye_client_ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/zendesk/sdk_faye_client_ios",
      "state" : {
        "revision" : "783d6d01d46922c1f6a2ea54efe17859499d98bb",
        "version" : "1.4.0"
      }
    },
    {
      "identity" : "sdk_http_client_ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/zendesk/sdk_http_client_ios",
      "state" : {
        "revision" : "0221b703f270eeb0072b89effd30dfec3d44960d",
        "version" : "0.12.0"
      }
    },
    {
      "identity" : "sdk_logger_ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/zendesk/sdk_logger_ios",
      "state" : {
        "revision" : "9717318fc76a78024ce758de6cb66359be9cd000",
        "version" : "0.7.0"
      }
    },
    {
      "identity" : "sdk_messaging_ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/zendesk/sdk_messaging_ios/",
      "state" : {
        "revision" : "75277c44afc363cf7823ae13ba950b2bf8cd9915",
        "version" : "2.12.0"
      }
    },
    {
      "identity" : "sdk_socket_client_ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/zendesk/sdk_socket_client_ios",
      "state" : {
        "revision" : "47d686da7b6319cf44a2492de126b305b3c809e7",
        "version" : "1.4.0"
      }
    },
    {
      "identity" : "sdk_storage_ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/zendesk/sdk_storage_ios",
      "state" : {
        "revision" : "19557da85ebaa8a04be610a314a6f7b8e7340790",
        "version" : "0.7.0"
      }
    },
    {
      "identity" : "sdk_ui_components_ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/zendesk/sdk_ui_components_ios",
      "state" : {
        "revision" : "669e032f3ba200e935df92078a14963bbfa29e0f",
        "version" : "2.7.0"
      }
    },
    {
      "identity" : "sdk_zendesk_ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/zendesk/sdk_zendesk_ios",
      "state" : {
        "revision" : "db1c2c5b95fb03084ee26b581a79329232ebbc72",
        "version" : "1.12.0"
      }
    }
  ],
  "version" : 2
}
```

I did some testing and you're correct with the above.

Could have been that xcode updated Core Utilities without updating Messaging and that caused the crash.

So:
Messaging 2.12.0 with CoreUtilities 1.4.0 crashes
Messaging 2.13.0 with CoreUtilities 1.4.0 is ok.

Thanks for you help, I'll close the issue.

**Comment by gaborvass-zendesk** [comment]
Closing this as it was fixed in [Release 2.13.1](https://developer.zendesk.com/documentation/zendesk-web-widget-sdks/sdks/ios/release_notes/#2131---2023-08-04)
